base-compat.cabal

```diff
@@ -22,8 +23,12 @@ library
       -Wall
   build-depends:
       base == 4.*
-    , ghc-prim
     , setenv
+
+  if impl(ghc >= 7.2) && impl(ghc < 7.6)
+    -- Prior to GHC 7.6, GHC.Generics lived in ghc-prim
+    build-depends: ghc-prim
```

Comment:
That's kind of the same reasoning that I apply to language extensions and CPP.  If a language extension is needed for some versions of base, but not harmful to have it for all versions of base, then I would not CPP it but just always enable it.